staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -375,19 +375,18 @@ func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.A
 	return &schema, nil
 }
 
-// OpenAPISchema fetches the open api schema using a rest client and parses the json.
-// Warning: this is very expensive (~1.2s)
-func (d *DiscoveryClient) OpenAPISchema() (*spec.Swagger, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger.json").Do().Raw()
+// OpenAPISchema fetches the open api schema using a rest client and parses the proto.
+func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
+	data, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
```

Comment:
> It still seems capricious and unnecessary to break. We are just talking about keeping the old code in the if err != nil block, right?

I would definitely not argue if that was it :-). 

We used to parse the json file into a `spec.Schema`, while the protobuf is parsed into a `openapi_v2.Document`. While there is code to go from `spec.Schema` to `openapi_v2.Document` (that code is used by the API-server to serve both json and protobuf), there is no code to go the other way around. That's why this PR is also updating every `spec.Schema` types to `openapi_v2.Document`.

I could still download the json file and convert it to protobuf and be happy, but to be honest, the json is 1.7mb (17x bigger than the gzipped protobuf), and the previous versions of the server don't support http caching, so it would have to be downloaded every-time, making it pretty much unusable anyway. 

In other words, I don't think kubectl of any version should ever use openapi with a pre-1.7 server version.